src/polytopes/segment.jl

```diff
@@ -138,16 +138,12 @@ function intersectpoint(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
   x1, x2 = vertices(s1)
   y1, y2 = vertices(s2)
 
-  # solve Ax = b
-  A = (y2 - y1) - (x2 - x1)
-  b = (x1 - y1)
-  s = zero(T)
-  for i in 1:length(A)
-    if !iszero(A[i])
-      s = b[i] / A[i]
-      break
-    end
-  end
+  tmp = y1 - y2
+
+  s = ((x1.coords[1] - y1.coords[1]) * tmp[2] -
+       (x1.coords[2] - y1.coords[2]) * tmp[1]) /
+      ((x1.coords[1] - x2.coords[1]) * tmp[2] -
+       (x1.coords[2] - x2.coords[2]) * tmp[1])
```

Comment:
It seems that the proposed formula only depends on point differences of the form (x1 - y1), (x1 - x2), ... In Meshes.jl you can obtain these differences as you did with `tmp = y1 - y2`. Please try to rewrite this expression so that all differences are computed a priori and then use the function `coordinates(y1 - y2)` to extract the corresponding coordinates in the reference system. We don't use `x.coords` because that is internal undocumented behavior. It is not part of the public API of points, makes sense?

After you fix the expression, make sure the PR also includes new tests for the case you reported in #29. This way we will never reintroduce the bug.

Reply:
I cleaned up the expression a little bit more following your advice and added the test case.